blueprints/controllers/ikea_e1812/ikea_e1812.yaml

```
@@ -59,197 +70,279 @@ blueprint:
         entity:
-          domain: input_text
-    helper_double_press_delay:
+          domain: input_text            
+    double_press_delay:
       name: (Optional) Helper - Double Press delay
-      description: Max delay between the first and the second button press for the double press event. Provide a value only if you are using a double press action. Increase this value if you notice that the double press action is not triggered properly.
-      default: 1000
+      description: Max delay between the first and the second button press for the
+        double press event. Provide a value only if you are using a double press action.
+        Increase this value if you notice that the double press action is not triggered
+        properly.
+      default: 500
       selector:
         number:
-          min: 0
-          max: 10000
+          min: 400.0
+          max: 5000.0
           unit_of_measurement: milliseconds
-          mode: box
-# Automation schema
+          mode: slider
+          step: 10.0
+    debounce_delay:
+      name: Helper - Debounce delay
+      description: Delay used for debouncing RAW controller events, when the device
+        is connected with an integration which needs debounce. Increase this value
+        only if the device is integrated with deCONZ, ZHA and you notice duplicate
+        action runs.
+      default: 200
+      selector:
+        number:
+          min: 0.0
+          max: 1000.0
+          unit_of_measurement: milliseconds
+          mode: slider
+          step: 10.0
 variables:
-  # convert input tags to variables, to be used in templates
-  integration: !input integration
-  button_long_loop: !input button_long_loop
-  button_double_press: !input button_double_press
-  helper_last_controller_event: !input helper_last_controller_event
-  # integration id used to select items in the action mapping
+  integration: !input 'integration'
+  button_long_loop: !input 'button_long_loop'
+  button_long_max_loops: !input 'button_long_max_loops'
+  button_double_press: !input 'button_double_press'
+  button_double_initial_press: !input 'button_double_initial_press'
+  action_button_long: !input 'action_button_long'
+  helper_last_controller_event: !input 'helper_last_controller_event'
+  double_press_delay: !input 'double_press_delay'
+  debounce_delay: !input 'debounce_delay'
+  # Subtracted debounce delay so that the timeout is really as set in double press delay but never go below minimum delay.
+  adjusted_double_press_delay: '{% if double_press_delay - debounce_delay < 100 %}100{% else %}
+                                {{ (double_press_delay - debounce_delay) | int }} {% endif %}'
   integration_id: '{{ integration | lower }}'
-  # mapping between actions and integrations
   actions_mapping:
     deconz:
-      button_short: [1002]
-      button_long: [1001]
-      button_release: [1003]
+      button_short:
+      - 1002
+      button_long:
+      - 1001
+      button_release:
+      - 1003
```

```suggestion
      button_short: [1002]
      button_long: [1001]
      button_release: [1003]
```

keep the inline format since it improves readability